consul/server.go

```diff
@@ -618,7 +622,18 @@ func (s *Server) setupRPC(tlsWrap tlsutil.DCWrapper) error {
 	// Provide a DC specific wrapper. Raft replication is only
 	// ever done in the same datacenter, so we can provide it as a constant.
 	wrapper := tlsutil.SpecificDC(s.config.Datacenter, tlsWrap)
-	s.raftLayer = NewRaftLayer(s.config.RPCSrcAddr, s.config.RPCAdvertise, wrapper)
+	tlsFunc := func(address raft.ServerAddress) bool {
+		s.localLock.RLock()
+		server, ok := s.localConsuls[address]
+		s.localLock.RUnlock()
+
+		if !ok {
+			return false
```

Comment:
This might be safer as return `true`, as if we don't know anything about the tag and defer to whether TLS is configured our not the old way (based on `tlsWrap`). Does that seem right?